web3/_utils/threads.py

```diff
@@ -91,23 +107,28 @@ def get(self, timeout=None):
 
 
 class TimerClass(threading.Thread):
-    def __init__(self, interval, callback, *args):
+    def __init__(self, interval: int, callback: Callable[..., Any], *args: Any) -> None:
         threading.Thread.__init__(self)
         self.callback = callback
         self.terminate_event = threading.Event()
         self.interval = interval
         self.args = args
 
-    def run(self):
+    def run(self) -> None:
         while not self.terminate_event.is_set():
             self.callback(*self.args)
             self.terminate_event.wait(self.interval)
 
-    def stop(self):
+    def stop(self) -> None:
         self.terminate_event.set()
 
 
-def spawn(target, *args, thread_class=ThreadWithReturn, **kwargs):
+def spawn(
+    target: Callable[..., Any],
```

Comment:
Same here, you could use `Callable[..., TReturn]` and have the function return type be `ThreadWithReturn[TReturn]`